### DISCUSSION TITLE
Migration/Restore of Nextcloud Instance

### DISCUSSION THREAD

**Discussion by Luatan** [nextcloud/all-in-one#4193] [Ideas] [answered]
I am using an external Backup Service to backup my Nextcloud instance. This gives me the ability to restore specific files/users to a point in time if needed. Since I needed to move servers I found that there is not an "official" way of doing this. Since I do not have a borg backup file it is not possible to just already available restore function.
I was able to workaround this by following your [migration guide](https://github.com/nextcloud/all-in-one/blob/main/migration.md). I wrote a script based on it. 

[restore.txt](https://github.com/nextcloud/all-in-one/files/14208513/restore.txt)

I considered to enable borg on the old instance to get the backup file and transfer it to the new server. But the problem with this is that it would take very long, since it will be a huge file. With my backup solution I can just restore to the new Server, but starting a new instance is a problem then.

Do you think a script for this migration would be a good idea to have in the repo? Ofcourse the script would need some more work.

**Comment by szaimen** [comment, marked as answer]
Hi, feel free to create a wiki article out of this. See https://github.com/nextcloud/all-in-one/discussions/new?category=wiki
